integration-tests/utils.ts

```diff
@@ -7,21 +7,76 @@ import * as AWS from 'aws-sdk';
 import axios, { AxiosInstance } from 'axios';
 import { Chance } from 'chance';
 import qs from 'qs';
+import { decode } from 'jsonwebtoken';
 import waitForExpect from 'wait-for-expect';
 
-export const getFhirClient = async (
-    role: 'auditor' | 'practitioner' = 'practitioner',
-    providedAccessToken?: string,
+const DEFAULT_TENANT_ID = 'tenant1';
+
+const getAuthParameters: (role: string, tenantId: string) => { PASSWORD: string; USERNAME: string } = (
+    role: string,
+    tenantId: string,
 ) => {
     const {
-        API_URL,
-        API_KEY,
-        API_AWS_REGION,
         COGNITO_USERNAME_PRACTITIONER,
         COGNITO_USERNAME_AUDITOR,
         COGNITO_PASSWORD,
-        COGNITO_CLIENT_ID,
+        COGNITO_USERNAME_PRACTITIONER_ANOTHER_TENANT,
+        MULTI_TENANCY_ENABLED,
     } = process.env;
+
+    if (COGNITO_USERNAME_PRACTITIONER === undefined) {
+        throw new Error('COGNITO_USERNAME_PRACTITIONER environment variable is not defined');
+    }
+    if (COGNITO_USERNAME_AUDITOR === undefined) {
+        throw new Error('COGNITO_USERNAME_AUDITOR environment variable is not defined');
+    }
+    if (COGNITO_PASSWORD === undefined) {
+        throw new Error('COGNITO_PASSWORD environment variable is not defined');
+    }
+
+    if (MULTI_TENANCY_ENABLED === 'true') {
+        if (COGNITO_USERNAME_PRACTITIONER_ANOTHER_TENANT === undefined) {
+            throw new Error('COGNITO_USERNAME_PRACTITIONER_ANOTHER_TENANT environment variable is not defined');
+        }
+    }
+
+    // for simplicity the different test users have the same password
+    const password = COGNITO_PASSWORD;
+    let username: string | undefined;
+    switch (role) {
+        case 'practitioner':
+            if (tenantId === undefined || tenantId === DEFAULT_TENANT_ID) {
+                username = COGNITO_USERNAME_PRACTITIONER;
+                break;
+            }
+            if (tenantId === 'tenant2') {
+                username = COGNITO_USERNAME_PRACTITIONER_ANOTHER_TENANT!;
+                break;
+            }
+            break;
+        case 'auditor':
+            username = COGNITO_USERNAME_AUDITOR;
+            break;
+        default:
+            break;
+    }
+
+    if (username === undefined) {
+        throw new Error('Could not find a username. Did you set up the integ tests correctly');
+    }
+
+    return {
+        USERNAME: username,
+        PASSWORD: password,
+    };
+};
+
+export const getFhirClient = async ({
+    role = 'practitioner',
+    providedAccessToken,
+    tenant = 'tenant1',
+}: { role?: 'auditor' | 'practitioner'; providedAccessToken?: string; tenant?: string } = {}) => {
```

Comment:
This is returning an `axios` instance not this body right?

Reply:
That's not the return type of the function. That's the type of the parameters(It gets a bit tricky due to the defaults and destructuring of params). The return type of the function is implicitly `Promise<AxiosInstance>`

```ts
export const getFhirClient = async ({
    role = 'practitioner',
    providedAccessToken,
    tenant = 'tenant1',
}: { role?: 'auditor' | 'practitioner'; providedAccessToken?: string; tenant?: string } = {}) => {
...
```

It could be specified as:
```ts
export const getFhirClient = async ({
    role = 'practitioner',
    providedAccessToken,
    tenant = 'tenant1',
}: { role?: 'auditor' | 'practitioner'; providedAccessToken?: string; tenant?: string } = {}): Promise<AxiosInstance> => {
...
```

Reply:
Ahh sorry yep that is right!